cmd/auth-handler.go

```diff
@@ -297,3 +369,55 @@ func (a authHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	writeErrorResponse(w, ErrSignatureVersionNotSupported, r.URL)
 }
+
+// isPutAllowed - check if PUT operation is allowed on the resource, this
+// call verifies bucket policies and IAM policies, supports multi user
+// checks etc.
+func isPutAllowed(atype authType, bucketName, objectName string, r *http.Request) (s3Err APIErrorCode) {
+	var accountName string
+	var owner bool
+	switch atype {
+	case authTypeUnknown:
+		return ErrAccessDenied
+	case authTypeSignedV2, authTypePresignedV2:
+		accountName, owner, s3Err = getReqAccessKeyV2(r)
+	case authTypeStreamingSigned, authTypePresigned, authTypeSigned:
+		region := globalServerConfig.GetRegion()
+		accountName, owner, s3Err = getReqAccessKeyV4(r, region)
+	}
+	if s3Err != ErrNone {
+		return s3Err
+	}
+
+	var claims = map[string]interface{}{}
+	s3Err = getClaimsFromToken(r, claims)
```

Comment:
This and the other place I commented are the only places where `getClaimsFromToken` seem to be called and they both send empty claims.

> it is needed to extract the claims.

What does this mean? Empty claims are sent into the function, verification is skipped, and only the token signature is verified. The name of the function is misleading, and it seems that it can be simpler.

Reply:
Empty claims are like a struct allocated we are unmarshalling JWT claims into this.